oauth_provider/tests/protocol.py

```diff
@@ -1,409 +1,409 @@
+            'oauth_callback': 'http://printer.example.com/request_token_ready',
+            'scope': 'photos', # custom argument to specify Protected Resource
+        }
+
+    def _make_access_token_parameters(self, token):
+        return {
+            'oauth_consumer_key': self.CONSUMER_KEY,
+            'oauth_token': token.key,
+            'oauth_signature_method': 'PLAINTEXT',
+            'oauth_signature': '%s&%s' % (self.CONSUMER_SECRET, token.secret),
+            'oauth_timestamp': str(int(time.time())),
+            'oauth_nonce': 'accessnonce',
+            'oauth_version': '1.0',
+            'oauth_verifier': token.verifier,
+            'scope': 'photos',
+        }
+
+    def _make_protected_access_parameters(self, access_token):
+        return {
+            'oauth_consumer_key': self.CONSUMER_KEY,
+            'oauth_token': access_token.key,
+            'oauth_signature_method': 'HMAC-SHA1',
+            'oauth_timestamp': str(int(time.time())),
+            'oauth_nonce': 'accessresourcenonce',
+            'oauth_version': '1.0',
+        }
+
+    def test_returns_invalid_params_empty_request(self):
+        """Printer website tries to access the photo and receives
+        HTTP 401 Unauthorized indicating it is private.
+        The Service Provider includes the following header with the response:
+        """
+        response = self.c.get("/oauth/request_token/")
+        self.assertEqual(response.status_code, 401)
+        self.assertEqual(response._headers['www-authenticate'], ('WWW-Authenticate', 'OAuth realm=""'))
+        self.assertEqual(response.content, 'Invalid request parameters.')
+
+    @skipIf(not hasattr(HttpResponse, 'reason_phrase'), 'Can\'t easilty test in prev versions')
+    def test_returns_correct_reason_phrase(self):
+        """For Django 1.6+ status phrase used to be 401 OK"""
+        response = self.c.get("/oauth/request_token/")
+        self.assertEqual(response.reason_phrase.upper(), 'UNAUTHORIZED')
+
+    def test_returns_401_wrong_callback(self):
+        #If you try to put a wrong callback, it will return an error
+        parameters = self._make_request_token_parameters()
+        parameters['oauth_callback'] = 'wrongcallback'
+        parameters['oauth_nonce'] = 'requestnoncewrongcallback'
+        response = self.c.get("/oauth/request_token/", parameters)
+        self.assertEqual(response.status_code, 401)
+        self.assertEqual(response.content, 'Invalid callback URL.')
+
+    def test_401_for_wrong_scope(self):
+        # If you try to access a resource with a wrong scope, it will return an error
+        parameters = self._make_request_token_parameters()
+        parameters['scope'] = 'videos'
+        parameters['oauth_nonce'] = 'requestnoncevideos'
+
+        response = self.c.get("/oauth/request_token/", parameters)
+
+        self.assertEqual(response.status_code, 401)
+        self.assertEqual(response.content.decode('utf-8'), 'Scope does not exist.')
```

Comment:
@mraarif ?

Reply:
for `python 3` the `bytes` output caused the test failure so converted it to `str`